typespec-extension/src/code-model-builder.ts

```diff
@@ -531,9 +531,29 @@ export class CodeModelBuilder {
     this.options["group-etag-headers"] = this.options["group-etag-headers"] ?? true;
 
     for (const client of clients) {
-      const codeModelClient = new CodeModelClient(client.name, this.getDoc(client.type), {
+      let clientName = client.name;
+      let javaNamespace = getJavaNamespace(this.namespace);
+      const clientFullName = client.name;
+      const clientNameSegments = clientFullName.split(".");
+      if (clientNameSegments.length > 1) {
+        clientName = clientNameSegments.at(-1)!;
+        javaNamespace = getJavaNamespace(
+          this.namespace + "." + clientNameSegments.slice(0, -1).join(".").toLowerCase(),
+        );
+      }
```

Comment:
break up the name e.g. `Contoso.Sub.ContosoClient`, so client name is `ContosoClient`, and `Contoso.Sub` get appended to default java namespace.

Reply:
Not sure how TCGC handles this. My guess is that it is same in the name.